src/main/java/org/spongepowered/api/event/world/chunk/ChunkEvent.java

```diff
@@ -54,25 +46,51 @@
     Vector3i getChunkPosition();
 
     /**
-     * Called before the {@link Chunk} is saved. Cancelling this will prevent any of
-     * the chunk's data being written to it's storage container.
+     * Called when a {@link Chunk} was unloaded.
      */
-    interface Pre extends SaveChunkEvent, Cancellable {
+    interface Unload extends ChunkEvent {
+
+    }
+
+    /**
+     * Called when a {@link Chunk} is performing a save.
+     */
+    interface Save extends ChunkEvent {
 
         /**
-         * Gets the {@link Chunk} being changed.
-         *
-         * @return The Chunk
+         * Called before the {@link Chunk} is saved. Cancelling this will prevent any of
+         * the chunk's data being written to it's storage container.
          */
-        Chunk getTargetChunk();
+        interface Pre extends Save, Cancellable {
+
+            /**
+             * Gets the {@link Chunk} being changed.
+             *
+             * @return The Chunk
+             */
+            Chunk getTargetChunk();
```

Comment:
If the chunk is available, is this a LevelChunk that has a live reference to the world?

Reply:
Could be ProtoChunk too I think.